kymatio/backend/torch_backend.py

```diff
@@ -3,163 +3,167 @@
-                axis removed.
-            """
-            ctx.p = 2
-            ctx.dim = -1
-            ctx.keepdim = False
-    
-            output = (x[...,0] * x[...,0] + x[...,1] * x[...,1]).sqrt()
-    
-            ctx.save_for_backward(x, output)
-    
-            return output
-    
-        @staticmethod
-        def backward(ctx, grad_output):
-            """Backward pass of the modulus
-    
-            This is a static method which does not require an instantiation of the
-            class.
-    
-            Arguments
-            ---------
-            ctx : context object
-                Collected during the forward pass. These are automatically added
-                by PyTorch and should not be touched. They are then used for the
-                backward pass.
-            grad_output : tensor
-                The gradient with respect to the output tensor computed at the
-                forward pass.
-    
-            Returns
-            -------
-            grad_input : tensor
-                The gradient with respect to the input.
-            """
-            x, output = ctx.saved_tensors
-            if ctx.dim is not None and ctx.keepdim is False and x.dim() != 1:
-                grad_output = grad_output.unsqueeze(ctx.dim)
-                output = output.unsqueeze(ctx.dim)
-    
-            grad_input = x.mul(grad_output).div(output)
-    
-            # Special case at 0 where we return a subgradient containing 0
-            grad_input.masked_fill_(output == 0, 0)
-    
-            return grad_input
-    
-    # shortcut for ModulusStable.apply
-    modulus = ModulusStable.apply
-    
-    class Modulus():
-        """This class implements a modulus transform for complex numbers.
-    
-            Usage
-            -----
-            modulus = Modulus()
-            x_mod = modulus(x)
-    
-            Parameters
-            ---------
-            x : tensor
-                Complex torch tensor.
-    
-            Returns
-            -------
-            output : tensor
-                A tensor with the same dimensions as x, such that output[..., 0]
-                contains the complex modulus of x, while output[..., 1] = 0.
-        """
-        def __call__(self, x):
-            complex_contiguous_check(x)
-            norm = modulus(x)[..., None]
-            return norm
-    
-    def complex_contiguous_check(self, x):
-        self.complex_check(x)
-        self.contiguous_check(x)
-    
+            A tensor with the same dimensions as x, such that output[..., 0]
+            contains the complex modulus of x, while output[..., 1] = 0.
+    """
+    def __call__(self, x):
+        complex_contiguous_check(x)
+        norm = modulus(x)[..., None]
+        return norm
+
+
+def input_checks(x):
+    if x is None:
+        raise TypeError('The input should be not empty.')
+
+    contiguous_check(x)
+
+
+def complex_check(x):
```

Comment:
Why are these not in the class?

Reply:
called in frontend

Reply:
Same here.

Reply:
Can't it get them from the backend? I agree that these functions belong more to the frontend, but they're currently organized into the backend, so let's keep them there.